tests/integration/case.js

```diff
@@ -0,0 +1,18 @@
+var assert = require('assert');
+var watchman = require('../../node');
+var client = new watchman.Client();
+var fs = require('fs');
+var path = require('path');
+
+var bar = fs.mkdtempSync('/tmp/BAR-');
+var foo = path.join(bar, 'FOO');
+fs.mkdirSync(foo);
+
+client.command(['watch', foo], function (error, resp) {
```

Comment:
the watch should be trying to watch `path.join(bar, 'foo')` so that we have a case mismatch.  Did you manage to get the error to trigger without making changes to the C files?